src/analysis/timeline/params.js

```diff
@@ -56,3 +57,10 @@ export function paramsFromJSON(
     weights: weightsFromJSON(p.weights),
   };
 }
+
+export function defaultParams(weights?: Weights): TimelineCredParameters {
```

Comment:
While I see your point about only weights being used for overrides. I feel like this approach (selective overrides, and a default parameters function) is somewhat unconventional. Most commonly I see optional arguments to a constructor or factory, which are then merged with the default values.

In some cases the defaults are also exposed, for example in the [yaml package](https://www.npmjs.com/package/yaml), because you may want to reuse your options object for multiple function calls. Though in this case, the `TimelineCred` class already acts as the way to store and reuse your options.

The bottom line is, I wouldn't make it the responsibility of the caller to know about `defaultParams` when constructing. Instead make it the constructor / factory's concern to fill in any parameters not provided. And not to be selective based on what's already used in the current code-base. As this convention will make exposing features in a library-style easier later on.